packages/cli/yaml/validator/src/rules/no-response-property/no-response-property.ts

```diff
@@ -0,0 +1,140 @@
+    name: "no-response-property",
+    create: ({ workspace }) => {
+        const typeResolver = new TypeResolverImpl(workspace);
+        return {
+            definitionFile: {
+                httpEndpoint: ({ endpoint }, { relativeFilepath, contents }) => {
+                    const { response } = endpoint;
+                    if (response == null) {
+                        return [];
+                    }
+                    const responseType = typeof response === "string" ? response : response.type;
+                    if (parseRawFileType(responseType) != null) {
+                        return [];
+                    } else if (parseRawTextType(responseType) != null) {
+                        return [];
+                    }
+                    const file = constructFernFileContext({
+                        relativeFilepath,
+                        definitionFile: contents,
+                        rootApiFile: workspace.definition.rootApiFile.contents,
+                        casingsGenerator: CASINGS_GENERATOR,
+                    });
+                    const responseBodyType = file.parseTypeReference(response);
+                    const responseProperty = typeof response !== "string" ? response.property : undefined;
+                    if (
+                        responseProperty != null &&
+                        !typeReferenceHasProperty(responseBodyType, responseProperty, file, typeResolver)
+                    ) {
+                        return [
+                            {
+                                severity: "error",
+                                message: `Response does not have a property named ${responseProperty}.`,
+                            },
+                        ];
+                    }
+                    return [];
+                },
+            },
+        };
+    },
+};
+
+function typeReferenceHasProperty(
+    typeReference: TypeReference,
+    property: string,
+    file: FernFileContext,
+    typeResolver: TypeResolver
+): boolean {
+    if (typeReference.type === "container" && typeReference.container.type === "optional") {
+        return typeReferenceHasProperty(typeReference.container.optional, property, file, typeResolver);
+    }
+    if (typeReference.type === "named") {
+        const resolvedType = typeResolver.resolveNamedTypeOrThrow({
+            referenceToNamedType: typeReference.name.originalName,
+            file,
+        });
+        switch (resolvedType._type) {
+            case "container":
+                return typeReferenceHasProperty(resolvedType.originalTypeReference, property, file, typeResolver);
+            case "named":
+                if (isRawObjectDefinition(resolvedType.declaration)) {
+                    return rawObjectSchemaHasProperty(resolvedType.declaration, property, file, typeResolver);
+                }
+                throw new Error("A custom response property is only supported for objects");
```

Comment:
@amckinney if we throw an error here this will cause all of the fern cli to stop and the user will get no output

Reply:
i think we should `return false` or the entire function should be updated to return an enum like: 
```
enum Result {
  ContainsProperty,
  DoesNotContainProperty,
  IsNotObject
}


```

Reply:
based on the enum the error message can be clearer too

Reply:
Much better, updated 👍